### DISCUSSION TITLE
Bang Olufsen IR sender

### DISCUSSION THREAD

**Discussion by Sill1000** [Arduino-IRremote/Arduino-IRremote#1131] [Q&A]
Hello

It's not clear for me... Does the ir_BangOlfsen.hpp sent IR codes at 455kHz or the ESP can only give a 38kHz modulation ?

Thanks

**Comment by ArminJo** [comment]
It is "or"

**Reply by Sill1000** [reply]
Sorry, but it's still not clear to me... 38 or 180kHz only ???
```

`	void IRsend::sendBangOlufsenRawDataLink(uint64_t aRawData, int_fast8_t aBits, bool aBackToBack, bool aUseDatalinkTiming) {
#if defined(USE_NO_SEND_PWM) || BEO_KHZ == 38 // BEO_KHZ == 38 is for unit test which runs the B&O protocol with 38 kHz instead 0f 455 kHz
    uint16_t tSendBEOMarkLength = aUseDatalinkTiming ? BEO_DATALINK_MARK : BEO_IR_MARK;

    /*
     * 455 kHz PWM is currently not supported, maximum is 180 kHz
     */
#if !defined(USE_NO_SEND_PWM)
    enableIROut (BEO_KHZ);
#endif`
```

**Reply by ArminJo** [reply]
455 kHz PWM is currently not supported, maximum is 180 kHz

**Reply by Sill1000** [reply]
But here:

[(https://github.com/Arduino-IRremote/Arduino-IRremote/discussions/1115#discussioncomment-5464805)]
you said it's possible to do 455kHz, no ?

**Reply by Sill1000** [reply]
I updated my latest post with the link!

**Reply by ArminJo** [reply]
Sorry, I also updated my comment. 
455 kHz is implemented for AVR timer2 but you are invited to implement it also for ESP.

**Reply by Sill1000** [reply]
Thanks ArminJo,
But unfortunately I don't have the knowledge to do this mods ! :-(
